_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,70 @@
+
+Waiting for a situation to become untenable is no-ones ideal scenario but we can find ways to mitigate the effects. A development team should be able to reach a consensus as to what are acceptable limits to change-lead-time and MTTR. If these limits are exceeded, it should be acceptable for the development team to focus on reducing them until they are within a tolerable range. Beyer, B. et al. (2016) offer a similar concept called "Error budgets". Specialist engineers are deployed to increase the reliability of critical services that have often been developed by other business units. These engineers assume on-call duties for the service provided that the service meets a minimum standard of stability. If this minimum standard is breached the engineers are within their rights to refuse support until the system can meet it. A development team could enter into a similar contract to relieve technical debt.
+
+As a body of software becomes larger it becomes difficult to maintain. This is aggravated by the inconvenient truth that software engineers are not tethered to their creations. They leave and new blood takes their place. New engineers need to understand the legacy left by the predecessors sans the education of having built the system in the first place. It is unreasonable to expect that a new engineer is going to be as effective as the seasoned veteran. We recognise nobody joins a new company with knowledge of it's internal processes, politics and three-letter-acryonyms, and so we should expect the same of it's software systems. Reducing the amount of learning required to be effective is a suitable means to ensuring that turnover does not effect the mean-time-to-repair and change-lead-time. This also provides the freedom required for specialisation.
+
+TODO: Image of communication links / Image
+
+Specialisation is commonly exploited. For a team of twenty engineers, sharing responsibility across twenty systems may feel like working on a crowded train. Drawing boundaries within the system and assigning responsibility for a particular subset of the group is how we commonly manage this. This works well provided that each smaller team still remains large enough to manage the burden of the subsystem. All we have actually done is (hopefully) distribute our engineers efficiently. Kepp in mind this redistribution occurs whether your team is working on microservices or a monolith. Any attempt to fight this is akin to trying to nail the wind to the ground, so you would be better off learning how to live with it. Managing your domain boundaries can be an effective way to reduce mean-time-to-repair and change-lead-time. This may not remove technical debt, but managing it becomes easier.
```

Comment:
> Kepp in mind
keep in mind

---
> Specialisation is commonly exploited

That is an intriguing claim and I am eager to learn more, but by the end of the paragraph, I'm not sure I did. 
</nickj>

Reply:
I don't think the claim is all that intriguing and is rather obvious?

There are two definitions of exploit when you chuck it into google;

1. make full use of and derive benefit from (a resource).
"500 companies sprang up to exploit this new technology"

2. make use of (a situation) in a way considered unfair or underhand.
"the company was exploiting a legal loophole"

The first definition is how I mean to use it - is the language too ambiguous?

Reply:
It wasn't obvious to me
- what in the paragraph was an example of specialisation 
- how specialisation was exploited
so in that sense it seemed like a frivolous statement. maybe if it said something like

> Specialisation is commonly exploited for good in high performing teams

and then qualifying it directly with one of your following statements such as

> [Effective teams] will draw boundaries within the system and take ownership and responsibility ....